daemon/analytics.c

```diff
@@ -0,0 +1,811 @@
+    freez(analytics_data.netdata_alarms_normal);
+    freez(analytics_data.netdata_alarms_warning);
+    freez(analytics_data.netdata_alarms_critical);
+    freez(analytics_data.netdata_charts_count);
+    freez(analytics_data.netdata_metrics_count);
+    freez(analytics_data.netdata_config_is_parent);
+    freez(analytics_data.netdata_config_hosts_available);
+    freez(analytics_data.netdata_host_cloud_available);
+    freez(analytics_data.netdata_host_aclk_available);
+    freez(analytics_data.netdata_host_aclk_implementation);
+    freez(analytics_data.netdata_host_agent_claimed);
+}
+
+/*
+ * Set a numeric/boolean data with a value
+ */
+void analytics_set_data (char **name, char *value) {
+    if (*name) freez(*name);
+    *name = strdupz(value);
+}
+
+/*
+ * Set a string data with a value
+ */
+void analytics_set_data_str (char **name, char *value) {
+    size_t value_string_len;
+    if (*name) freez(*name);
+    value_string_len = strlen(value) + 3;
+    *name = mallocz(sizeof(char) * value_string_len);
+    snprintfz(*name, value_string_len, "\"%s\"", value);
+}
+
+/*
+ * Get data, used by web api v1
+ */
+void analytics_get_data (char *name, BUFFER *wb) {
+    buffer_strcat(wb, name);
+}
+
+/*
+ * Log hits on the allmetrics page, with prometheus parameter
+ */
+void analytics_log_prometheus(void) {
+    if (likely(analytics_data.prometheus_hits < ANALYTICS_MAX_PROMETHEUS_HITS)) {
+        analytics_data.prometheus_hits++;
+        char b[7];
+        snprintfz(b, 6, "%d", analytics_data.prometheus_hits);
+        analytics_set_data (&analytics_data.netdata_allmetrics_prometheus_used, b);
+    }
+}
+
+/*
+ * Log hits on the allmetrics page, with shell parameter (or default)
+ */
+void analytics_log_shell(void) {
+    if (likely(analytics_data.shell_hits < ANALYTICS_MAX_SHELL_HITS)) {
+        analytics_data.shell_hits++;
+        char b[7];
+        snprintfz(b, 6, "%d", analytics_data.shell_hits);
+        analytics_set_data (&analytics_data.netdata_allmetrics_shell_used, b);
+    }
+}
+
+/*
+ * Log hits on the allmetrics page, with json parameter
+ */
+void analytics_log_json(void) {
+    if (likely(analytics_data.json_hits < ANALYTICS_MAX_JSON_HITS)) {
+        analytics_data.json_hits++;
+        char b[7];
+        snprintfz(b, 6, "%d", analytics_data.json_hits);
+        analytics_set_data (&analytics_data.netdata_allmetrics_json_used, b);
+    }
+}
+
+/*
+ * Log hits on the dashboard, (when calling HELLO).
+ */
+void analytics_log_dashboard(void) {
+    if (likely(analytics_data.dashboard_hits < ANALYTICS_MAX_DASHBOARD_HITS)) {
+        analytics_data.dashboard_hits++;
+        char b[7];
+        snprintfz(b, 6, "%d", analytics_data.dashboard_hits);
+        analytics_set_data (&analytics_data.netdata_dashboard_used, b);
+    }
+}
+
+void analytics_mirrored_hosts( void ) {
+    RRDHOST *host;
+    int count = 0;
+    int reachable = 0;
+    int unreachable = 0;
+    char b[10];
```

Comment:
Assuming that we will never have 2147483647 `count`, `reachable` and `unreachable`.  The size `10` is enough to store data, but to work with all possible integer values, we would need one more byte in this vector.

Reply:
Thank you my friend! Did it `11` and changed from `9` -> `10` the limits in the later lines.